### PR TITLE
Helpful runtime error message in pgindent target when there's no objdump

### PR DESCRIPTION
**Pull request by amytai**
Trying to run `make pgindent` should result in runtime error if OBJDUMP not installed. This is in contrast to the current error message, which is easily missed during compile-time.

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -153,14 +153,18 @@ else ()
...
+if (PGINDENT)
+  if (OBJDUMP)
+    add_custom_command(OUTPUT typedefs.list
+      DEPENDS ${PROJECT_NAME}
+      COMMAND sh ${CMAKE_BINARY_DIR}/scripts/generate_typedefs.sh > typedefs.list)
+    add_custom_target(pgindent
+      COMMAND ${PGINDENT} -typedefs typedefs.list -code-base ${CMAKE_SOURCE_DIR}/src
+      DEPENDS typedefs.list)
+  else()
+	message(STATUS "Found pgindent, but did not have objdump installed. Cannot use pgindent")
```
Comment (RobAtticus) [inline src/CMakeLists.txt]: "Found pgindent, but objdump not installed - cannot use pgindent." 

Reads a bit better to me anyway.

Comment (RobAtticus) [reply src/CMakeLists.txt]: Or perhaps to not bury the lede:
"Cannot use pgindent - found pgindent, but objdump not installed"

Comment (amytai) [reply src/CMakeLists.txt]: Like the second. Done.

**Comment by mfreed** [comment]
Please squash the commits in the PR...

**Comment by amytai** [comment]
Done

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -153,14 +153,18 @@ else ()
...
+if (PGINDENT)
+  if (OBJDUMP)
+    add_custom_command(OUTPUT typedefs.list
+      DEPENDS ${PROJECT_NAME}
+      COMMAND sh ${CMAKE_BINARY_DIR}/scripts/generate_typedefs.sh > typedefs.list)
+    add_custom_target(pgindent
+      COMMAND ${PGINDENT} -typedefs typedefs.list -code-base ${CMAKE_SOURCE_DIR}/src
+      DEPENDS typedefs.list)
+  else()
+	message(STATUS "Cannot use pgindent -- found pgindent, but objdump not installed.")
```
Comment (erimatnor) [inline src/CMakeLists.txt]: So, we already have an error if `OBJDUMP` is not set in `scripts/CMakeLists.txt`. Not sure how this one improves on that?

I think what we may want to do here is to install a `pgindent` make target also in the negative case, which outputs an informative message when you do `make pgindent`. I think such a runtime error is more helpful.

Comment (amytai) [reply src/CMakeLists.txt]: Oh wow, I completely missed the `scripts/CMakeLists.txt` message. You're right, this means this compile-time error will also be missed. Added runtime error instead.

File: scripts/generate_typedefs.sh.in [hunk 3]
```diff
@@ -1,16 +1,23 @@
 #!/bin/bash
 
-TMPDIR=${TMPDIR:-$(mktemp -d 2>/dev/null || mktemp -d -t 'timescaledb_typedef')}
+hasObjdump=@OBJDUMP@
```
[the remark this answers is not in the document]
Comment (amytai) [reply scripts/generate_typedefs.sh.in]: Fixed.

**Comment by erimatnor** [comment]
@amytai Btw, add a description to your commit, even if short.

**Pull request merged**